### PR TITLE
Proteinortho

### PR DESCRIPTION
**Pull request by pmjklemm**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source proteinortho`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test proteinortho`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict proteinortho` (after doing `brew install <formula>`)?

-----

`brew audit --strict proteinortho` fails with the following message:

```
An error occurred while installing nokogiri (1.10.3), and Bundler cannot continue.
Make sure that `gem install nokogiri -v '1.10.3' --source 'https://rubygems.org/'` succeeds before bundling.
```

But `brew audit --strict nokogiri` fails too (even without proteinortho installed). The installation of nokogiri fails too `sudo gem install nokogiri -v '1.10.3' --source 'https://rubygems.org/'` with 
> ld: warning: ignoring file /usr/local/Cellar/xz/5.2.4/lib/liblzma.dylib, file was built for x86_64 which is not the architecture being linked (i386): /usr/local/Cellar/xz/5.2.4/lib/liblzma.dylib
> extconf failed, exit code 1

I am not sure if this XML parser is neccessary for my program ?

### CODE REVIEW

File: Formula/proteinortho.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+class Proteinortho < Formula
+  desc "Proteinortho is a tool to detect orthologous genes within different species"
+  homepage "https://gitlab.com/paulklemm_PHD/proteinortho"
+  url "https://gitlab.com/paulklemm_PHD/proteinortho/-/archive/v6.0b/proteinortho-v6.0b.tar.gz"
+  sha256 "f59c436f6461f318ccebc1177a67eb70c29ef9ba222ad724de1fb2a769a120b5"
+  depends_on "cmake" => :build
+  depends_on "gcc"
```
Comment (SMillerDev) [inline Formula/proteinortho.rb]: does it depend on gcc at runtime?

File: Formula/proteinortho.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  # get the latest g++ compiler from brew (Cellar)
+  CXX = `ls /usr/local/Cellar/gcc/*/bin/g++* | sort | tail -n1 | tr -d '\n'`
```
Comment (SMillerDev) [inline Formula/proteinortho.rb]: this is not okay, the `/usr/local` install path isn't guaranteed and neither is the shell. You can use an `ENV.cxx11` call or something similar if you need a specific version.

File: Formula/proteinortho.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+class Proteinortho < Formula
+  desc "Proteinortho is a tool to detect orthologous genes within different species"
+  homepage "https://gitlab.com/paulklemm_PHD/proteinortho"
+  url "https://gitlab.com/paulklemm_PHD/proteinortho/-/archive/v6.0b/proteinortho-v6.0b.tar.gz"
```
Comment (SMillerDev) [inline Formula/proteinortho.rb]: This is marked as a beta release, we don't ship unstable versions: https://docs.brew.sh/Acceptable-Formulae#stable-versions

File: Formula/proteinortho.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  sha256 "f59c436f6461f318ccebc1177a67eb70c29ef9ba222ad724de1fb2a769a120b5"
+  depends_on "cmake" => :build
+  depends_on "gcc"
+
+  # get the latest g++ compiler from brew (Cellar)
+  CXX = `ls /usr/local/Cellar/gcc/*/bin/g++* | sort | tail -n1 | tr -d '\n'`
+
+  def install
+    system "make", "clean"
```
Comment (SMillerDev) [inline Formula/proteinortho.rb]: Is this needed? The install will always be done in a clean directory.

**Review by SMillerDev — CHANGES_REQUESTED** [review]
Thanks for the pull request, there's still a couple of comments before we can accept it into homebrew-core.

File: Formula/proteinortho.rb (hunk 1 above)
Comment (pmjklemm) [reply Formula/proteinortho.rb]: It is it linked dynamically, that is why i thought i need that

File: Formula/proteinortho.rb (hunk 2 above)
Comment (pmjklemm) [reply Formula/proteinortho.rb]: ahhh ok i was searching for that, i will fix that

File: Formula/proteinortho.rb (hunk 3 above)
Comment (pmjklemm) [reply Formula/proteinortho.rb]: I declared it as beta, since i did not published it. But is tested intensively and stable. I will remove the b.

File: Formula/proteinortho.rb (hunk 4 above)
Comment (pmjklemm) [reply Formula/proteinortho.rb]: You are right, that is not needed

File: Formula/proteinortho.rb (hunk 2 above)
Comment (pmjklemm) [reply Formula/proteinortho.rb]: So i tried `ENV.cxx11` but it does not seem to have any effect (the g++ compiler is still the apple one and not the Cellar/brew g++)
After calling `ENV.cxx11` what happens

```
  def install
    ENV.cxx11
    system "mkdir", "#{prefix}/bin"
    system "make", "install", "PREFIX=#{prefix}/bin"
  end
```
Then it fails because apples g++ does not have gfortran...
Any clue what i am doing wrong? 
(I looked at /usr/bin/g++ and it is still the same old clang g++ and $CXX points to clang++, that is not what i need)

Comment (SMillerDev) [reply Formula/proteinortho.rb]: Here's a formula with the same problem (needing gfortran) https://github.com/Homebrew/homebrew-core/blob/master/Formula/octave.rb

Comment (pmjklemm) [reply Formula/proteinortho.rb]: Similar to octave i have now this solution:
```
  def install
    system "mkdir", "#{prefix}/bin"
    ENV["CXX"] = `ls #{Formula["gcc"].opt_bin.realpath}/g++-* | sort | tail -n1`.chomp
    system "make", "CPP=$CXX" , "install", "PREFIX=#{prefix}/bin"
  end
```
Would that be ok?

**Comment by pmjklemm** [comment]
now i fixed these:
>   Dependency 'lapack' may be unnecessary as it is provided by macOS; try to build this formula without it.

and:
>   * C: 6: col 31: Trailing whitespace detected.
>   * C: 7: col 36: Trailing whitespace detected.
>   * C: 8: col 3: dependency "lapack" (line 8) should be put before dependency "pkg-config" (line 7)
>   * C: 8: col 32: Trailing whitespace detected.
>   * C: 9: col 3: dependency "gcc" (line 9) should be put before dependency "pkg-config" (line 7)
>   * C: 9: col 29: Trailing whitespace detected.
>   * C: 12: col 13: Use the `mkdir` Ruby method instead of `system "mkdir", "#{prefix}/bin"`
>   * C: 12: col 32: "#{prefix}/bin" should be "#{bin}"
>   * C: 15: col 78: Space found before comma.
>   * C: 15: col 110: "#{prefix}/bin" should be "#{bin}"
>   * Non-executables were installed to "/usr/local/opt/proteinortho/bin"
>     The offending files are:
>       /usr/local/opt/proteinortho/bin/grepfasta.pl

**Comment by pmjklemm** [comment]
seems not to work, since now it does not find g++-8 when --build-bottle
`
g++-8: error: : No such file or directory
`

**Comment by sjackman** [comment]
`proteinortho` version 5.15 is found in Brewsci/science.
https://github.com/brewsci/homebrew-science/blob/master/Formula/proteinortho.rb

This version did not require Fortran and did not require GCC. Does this new version 6 require Fortran?

```ruby
ENV["CXX"] = `ls #{Formula["gcc"].opt_bin.realpath}/g++-* | sort | tail -n1`.chomp
```
Is it strictly necessary to use GNU gcc to compile the C++ code? Can `gfortran` be used to compile the Fortran code, and `clang++` be used to compile the C++ code?

If GCC is required, remove the above `ENV["CXX"] = …` line and add…
```ruby
fails_with :clang # needs fortran
```

File: Formula/proteinortho.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
+class Proteinortho < Formula
+  desc "Detecting orthologous genes within different species"
+  homepage "https://gitlab.com/paulklemm_PHD/proteinortho"
+  url "https://gitlab.com/paulklemm_PHD/proteinortho/-/archive/v6.0/proteinortho-v6.0.tar.gz"
+  sha256 "e82522cdae62c182fc3bb75903e8ce3a388573fccbea16b72a0a8c47a653a34b"
```
Comment (sjackman) [inline Formula/proteinortho.rb]:
```suggestion
  sha256 "e82522cdae62c182fc3bb75903e8ce3a388573fccbea16b72a0a8c47a653a34b"

```
Add a blank line after `sha256`.

File: Formula/proteinortho.rb [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "lapack"
```
Comment (sjackman) [inline Formula/proteinortho.rb]: @dpo Can OpenBLAS be used rather than LAPACK? Do you have a preference?

File: Formula/proteinortho.rb [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  depends_on "gcc" # for gfortran
+
```
Comment (sjackman) [inline Formula/proteinortho.rb]:
```suggestion

  fails_with :clang # needs fortran

```

File: Formula/proteinortho.rb [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+  def install
+    mkdir bin.to_s
+    ENV["LIBRARY_PATH"] = `pkg-config --libs --static lapack`.chomp
+    ENV["CXX"] = `ls #{Formula["gcc"].opt_bin.realpath}/g++-* | sort | tail -n1`.chomp
```
Comment (sjackman) [inline Formula/proteinortho.rb]:
```suggestion
```

File: Formula/proteinortho.rb [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+  url "https://gitlab.com/paulklemm_PHD/proteinortho/-/archive/v6.0/proteinortho-v6.0.tar.gz"
+  sha256 "e82522cdae62c182fc3bb75903e8ce3a388573fccbea16b72a0a8c47a653a34b"
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "lapack"
+  depends_on "gcc" # for gfortran
+
+  def install
+    mkdir bin.to_s
+    ENV["LIBRARY_PATH"] = `pkg-config --libs --static lapack`.chomp
```
Comment (sjackman) [inline Formula/proteinortho.rb]:
```suggestion
```
Is this line needed? What's the error without it?

File: Formula/proteinortho.rb [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+    ENV["CXX"] = `ls #{Formula["gcc"].opt_bin.realpath}/g++-* | sort | tail -n1`.chomp
+    system "make", "CXX=#{ENV["CXX"]}", "CXXLIBRARY='#{ENV["LIBRARY_PATH"]}'", "install", "PREFIX=#{bin}"
```
Comment (sjackman) [inline Formula/proteinortho.rb]:
```suggestion
    system "make", "install", "PREFIX=#{bin}"
```

File: Formula/proteinortho.rb (hunk 6 above)
Comment (pmjklemm) [reply Formula/proteinortho.rb]: i need the dsyevr/dsyevx function of lapack, i am not sure if that is also part of BLAS.

File: Formula/proteinortho.rb (hunk 5 above)
Comment (pmjklemm) [reply Formula/proteinortho.rb]: I added a blank in the next commit. The 9c18538, just fixes the file permission.

File: Formula/proteinortho.rb (hunk 7 above)
Comment (pmjklemm) [reply Formula/proteinortho.rb]: Then it fails with 
No package 'lapack' found
also i need `pkg-config` to find lapack

File: Formula/proteinortho.rb (hunk 10 above)
Comment (pmjklemm) [reply Formula/proteinortho.rb]: Thank you so much, this just works ! I will upload a new version soon

```

  depends_on "cmake" => :build
  fails_with :clang # needs fortran

  def install
    mkdir bin.to_s
    system "make", "install", "PREFIX=#{bin}"
  end

```

**Comment by pmjklemm** [comment]
00403ff : added manual.html and doc.install "manual.html" in the ruby script.

So since there is allready a old proteinortho in homebrew-science, should i now make a pull request in homebrew-science? (I thought homebrew-science is deprecated)

File: Formula/proteinortho.rb (hunk 1 above)
Comment (zbeekman) [reply Formula/proteinortho.rb]: If there is Fortran in the sources, then `depends_on "gcc"` is the correct way to enable Fortran compilation. Dynamic linkage and seeing dependencies on gcc runtime libs when you build on your machine with, e.g., `export CC=gcc-8 ; cmake .. ; ... ` doesn't necessarily mean you have a true dependency on gcc.

TL;DR: Unless you are building Fortran sources, or have a legitimate reason that you can't build with Apple Clang, then I don't think you need the GCC dependency.

File: Formula/proteinortho.rb (hunk 6 above)
Comment (zbeekman) [reply Formula/proteinortho.rb]: We should move to OpenBLAS for new formulae, see #39226

**Comment by zbeekman** [comment]
Hi guys, I'll look at brewsci/bio, and the upstream, but I am really not convinced that we need to or should be building with g++ over clang++. I don't see a clear reason stated in the conversation above.

File: Formula/proteinortho.rb (hunk 7 above)
Comment (zbeekman) [reply Formula/proteinortho.rb]: Hi guys, I'm a bit confused here: do the C/C++ portions of the code fail with Apple clang? If so, why? "needs Fortran" is not a valid reason for adding a `fails_with` block...

Generally, when formulae require Fortran, you add a `depends_on "gcc"` and that is sufficient. All C code is generally built with Apple Clang, and Fortran with the latest GCC.

Also, if you need pkg-config to find lapack in a homebrew formula, for a CMake based project then, you're doing something wrong somewhere...

**Review by zbeekman — CHANGES_REQUESTED** [review]
I'm going to take a look at upstream and see if I can clean things up and make this formula sane again...

File: Formula/proteinortho.rb (hunk 6 above)
Comment (sjackman) [reply Formula/proteinortho.rb]: That's my understanding (but I'm not an expert).

File: Formula/proteinortho.rb (hunk 7 above)
Comment (sjackman) [reply Formula/proteinortho.rb]: I agree with this assessment. Compile the Fortran code with `gfotran` and the C++ code with `clang++`.

File: Formula/proteinortho.rb (hunk 10 above)
Comment (sjackman) [reply Formula/proteinortho.rb]:
```ruby
fails_with :clang # needs fortran
```
Is this line needed? What's the error message without it?

**Comment by sjackman** [comment]
> So since there is allready a old proteinortho in homebrew-science, should i now make a pull request in homebrew-science? (I thought homebrew-science was deprecated)

Yes, Brewsci/science is legacy. It only exists as a record of what was. The only PRs accepted into Brewsci/science are to migrate formulae to other taps.

A PR to Brewsci/science is needed _after_ this PR is merged. Run…
```sh
brew update
brew tap linuxbrew/developer
brew tap brewsci/science
brew migrate-formula --tap=homebrew/core brewsci/science/proteinortho
```

File: Formula/proteinortho.rb [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+  desc "Detecting orthologous genes within different species"
+  homepage "https://gitlab.com/paulklemm_PHD/proteinortho"
+  url "https://gitlab.com/paulklemm_PHD/proteinortho/-/archive/v6.0/proteinortho-v6.0.tar.gz"
+  sha256 "1df4460f7cd5654c1e0c0c7abf546b50288265c895b1bf28dde618d7f080e805"
+
+  depends_on "cmake" => :build
+  fails_with :clang # needs fortran
+
+  def install
+    mkdir bin.to_s
```
Comment (sjackman) [inline Formula/proteinortho.rb]:
```suggestion
    bin.mkpath
```

**Review by sjackman — CHANGES_REQUESTED** [review]

**Comment by pmjklemm** [comment]
> Why did the sha256 change?

E.g. file permission failed in the automatic audit test, ( i forgot to chomd +x a file). And i found out that `-WL,-whole-archive` only works on linux, so i had to change this in the makefile for osx.

> Hi guys, I'll look at brewsci/bio, and the upstream, but I am really not convinced that we need to or should be building with g++ over clang++. I don't see a clear reason stated in the conversation above.

I dont get clang++ to work with gfortran, thats why i require gcc. If you can give me a hint how clang++ and fortran can work together, i will change that.

`fails_with :clang` replaced with:
- `depends_on "gfortran"` and i explicitly pass CXX=clang++ to make, then it fails with `library not found for -lgfortran`.
- `depends_on "gcc"` and CXX=clang++ -> `library not found for -lgfortran`.
- `depends_on "gcc"` and CXX=g++ -> `library not found for -lgfortran`.

**Comment by sjackman** [comment]
>> Why did the sha256 change?

> E.g. file permission failed in the automatic audit test, ( i forgot to chomd +x a file). And i found out that -WL,-whole-archive only works on linux, so i had to change this in the makefile for osx.

Please bump the version number whenever the upstream tarball is changed. See https://semver.org

**Comment by sjackman** [comment]
> I dont get clang++ to work with gfortran, thats why i require gcc. If you can give me a hint how clang++ and fortran can work together, i will change that.

> fails_with :clang replaced with:
> 
> depends_on "gcc" and CXX=clang++ -> library not found for -lgfortran.
> depends_on "gcc" and CXX=g++ -> library not found for -lgfortran.
> (I need gfortran to link against lapack)

Are you linking C++ and Fortran code into one executable?

I don't believe you're meant to explicitly link against `-lgfortran`. It's implied when you link using `gfortran`.

Try using `-lopenblas` rather than `-llapack`.

**Comment by pmjklemm** [comment]
@sjackman Thank you so much! I was not awaire that -lopenblas inherits lapack, but now it works without any dependancy. (Very weird, that lapack needs gfortran for linking but the "parent" openblas not)

**Review by sjackman — APPROVED** [review]

**Comment by pmjklemm** [comment]
ping @zbeekman @SMillerDev 

I think all requested changes are now implemented. 
I just need to do
```
brew update
brew tap linuxbrew/developer
brew tap brewsci/science
brew migrate-formula --tap=homebrew/core brewsci/science/proteinortho
``` 
after mergin?

**Comment by zbeekman** [comment]
I’ll take another look. Honestly, I have concerns about the upstream build
system and it’s impact on maintainability. It appears to rely heavily on
custom Perl scripts and other non-canonical techniques but I’ll take
another look later today.

**Comment by zbeekman** [comment]
@pmjklemm Let's hold off on worrying about linuxbrew and brewsci for the moment, and address issues relevant to Homebrew-core here.

After further review, these changes look good to me. The build system is sufficiently simple that I think the maintenance burden won't be excessive if we include this.

A final question for @pmjklemm: Do we want/need a dependency on perl? I notice the build system is heavily dependent on perl. MacOS 10.14 ships with system perl v5.18.2 and homebrew ships perl v5.28.1.

So this question essentially has 2 parts:

1. Should we add a dependency on brewed perl?
2. If so should it be a build dependency only, or a runtime dependency too?

CI is green for OSX, so this question is probably more relevant for the eventual downstream linuxbrew-core merge.

**Review by zbeekman — APPROVED** [review]

**Comment by pmjklemm** [comment]
@zbeekman
> @pmjklemm Let's hold off on worrying about linuxbrew and brewsci for the moment, and address issues relevant to Homebrew-core here.
> 
> After further review, these changes look good to me. The build system is sufficiently simple that I think the maintenance burden won't be excessive if we include this.
> 
> A final question for @pmjklemm: Do we want/need a dependency on perl? I notice the build system is heavily dependent on perl. MacOS 10.14 ships with system perl v5.18.2 and homebrew ships perl v5.28.1.
> 
> So this question essentially has 2 parts:
> 
> 1. Should we add a dependency on brewed perl?
> 2. If so should it be a build dependency only, or a runtime dependency too?
> 
> CI is green for OSX, so this question is probably more relevant for the eventual downstream linuxbrew-core merge.

So my main system runs on perl 5.26.2, but i also tested under 5.18.2 and it works just fine. 
Perl is only used at runtime, for building perl is not needed and brewperl is also not needed as far as i can tell.

**Comment by zbeekman** [comment]
> So my main system runs on perl 5.26.2, but i also tested under 5.18.2 and it works just fine.
> Perl is only used at runtime, for building perl is not needed and brewperl is also not needed as far as i can tell.

If system perl works fine then I'm happy to keep things as is.

I may be mistaken, but, looking at the makefile, I see a log of perl scripts... maybe they're just being moved around and/or written, but my quick once over left me with the impression that it was a build-time dep too... but if we can just use system perl on macOS and it doesn't cause problems for linux then I think this is ready to ship.

**Comment by sjackman** [comment]
@pmjklemm Congrats on getting this PR merged! Thank you for your contribution to Homebrew/core.

You can run the commands in https://github.com/Homebrew/homebrew-core/pull/39428#issuecomment-488347829 to remove this formula from Brewsci/science now and mark it as having migrated to Homebrew/core.

**Comment by pmjklemm** [comment]
> I may be mistaken, but, looking at the makefile, I see a log of perl scripts... maybe they're just being moved around and/or written, but my quick once over left me with the impression that it was a build-time dep too... but if we can just use system perl on macOS and it doesn't cause problems for linux then I think this is ready to ship.

The perl scripts are just a moved around.

---

Thank you all.

**Comment by sjackman** [comment]
Fork `brewsci/science`:
```sh
brew update
brew tap linuxbrew/developer
brew tap brewsci/science
brew install hub
cd $(brew --repo brewsci/science)
hub fork
brew migrate-formula --tap=homebrew/core brewsci/science/proteinortho
```

**Comment by pmjklemm** [comment]
@sjackman not sure if that is working?

```
(base) [pmjk@pc:homebrew-science]$ brew migrate-formula --tap=homebrew/core brewsci/science/proteinortho
Warning: Use proteinortho instead of deprecated brewsci/science/proteinortho
Warning: proteinortho is already in homebrew/core
```

But i got no further message

**Comment by sjackman** [comment]
You have to undo the previous unsuccessful migration that failed due to the missing fork. If you have no important uncommitted work in any tap you can run `brew update-reset`. If you want to be more selective, run…
```sh
cd $(brew --repo brewsci/science)
git fetch origin
git checkout master
git reset --hard origin/master
```
then run
```sh
cd $(brew --repo brewsci/science)
brew migrate-formula --tap=homebrew/core brewsci/science/proteinortho
```

**Comment by pmjklemm** [comment]
@sjackman 

using `brew update-reset` and the lower part:
```
==> Migrating brewsci/science/proteinortho to homebrew/core
Warning: Skipping new formula PR because formula already exists: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/proteinortho.rb
Already on 'master'
Your branch is up to date with 'origin/master'.
Already up to date.
fatal: A branch named 'migrate-proteinortho' already exists.
Error: Failure while executing; `git checkout -b migrate-proteinortho` exited with 128.
```

with the upper part i get:

```
Warning: Skipping new formula PR because formula already exists: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/proteinortho.rb
Already on 'master'
Your branch is up to date with 'origin/master'.
Already up to date.
Switched to a new branch 'migrate-proteinortho'
rm 'Formula/proteinortho.rb'
[migrate-proteinortho 4da9029] proteinortho: migrate to homebrew/core
 2 files changed, 1 insertion(+), 35 deletions(-)
 delete mode 100644 Formula/proteinortho.rb
fatal: 'paulklemm' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Error: Failure while executing; `git push paulklemm migrate-proteinortho` exited with 128.
```

**Comment by sjackman** [comment]
```
fatal: 'paulklemm' does not appear to be a git repository
```

Did the `hub fork` succeed?

```sh
set -x
cd $(brew --repo brewsci/science)
git checkout master
git branch -D migrate-proteinortho
git fetch origin
git reset --hard origin/master
hub fork
git fetch paulklemm
brew migrate-formula --tap=homebrew/core brewsci/science/proteinortho
```

**Comment by pmjklemm** [comment]
@sjackman so `set -x` destroys my shell (?) all commands will then only produce the same garbage:

```
++ url_path+=m
++ (( ++i ))
++ (( i < 57 ))
++ ch=e
++ [[ e =~ [/._~A-Za-z0-9-] ]]
++ url_path+=e
++ (( ++i ))
++ (( i < 57 ))
++ ch=b
++ [[ b =~ [/._~A-Za-z0-9-] ]]
++ url_path+=b
++ (( ++i ))
++ (( i < 57 ))
++ ch=r
++ [[ r =~ [/._~A-Za-z0-9-] ]]
++ url_path+=r
++ (( ++i ))
++ (( i < 57 ))
++ ch=e
++ [[ e =~ [/._~A-Za-z0-9-] ]]
++ url_path+=e
++ (( ++i ))
++ (( i < 57 ))
++ ch=w
++ [[ w =~ [/._~A-Za-z0-9-] ]]
++ url_path+=w
++ (( ++i ))
++ (( i < 57 ))
++ ch=-
++ [[ - =~ [/._~A-Za-z0-9-] ]]
++ url_path+=-
++ (( ++i ))
++ (( i < 57 ))
++ ch=s
++ [[ s =~ [/._~A-Za-z0-9-] ]]
++ url_path+=s
++ (( ++i ))
++ (( i < 57 ))
++ ch=c
++ [[ c =~ [/._~A-Za-z0-9-] ]]
++ url_path+=c
++ (( ++i ))
++ (( i < 57 ))
++ ch=i
++ [[ i =~ [/._~A-Za-z0-9-] ]]
++ url_path+=i
++ (( ++i ))
++ (( i < 57 ))
++ ch=e
++ [[ e =~ [/._~A-Za-z0-9-] ]]
++ url_path+=e
++ (( ++i ))
++ (( i < 57 ))
++ ch=n
++ [[ n =~ [/._~A-Za-z0-9-] ]]
++ url_path+=n
++ (( ++i ))
++ (( i < 57 ))
++ ch=c
++ [[ c =~ [/._~A-Za-z0-9-] ]]
++ url_path+=c
++ (( ++i ))
++ (( i < 57 ))
++ ch=e
++ [[ e =~ [/._~A-Za-z0-9-] ]]
++ url_path+=e
++ (( ++i ))
++ (( i < 57 ))
++ printf '\e]7;%s\a' file://pk.local/usr/local/Homebrew/Library/Taps/brewsci/homebrew-science
++ '[' -w . ']'
++ echo -n 32
```
So without the set -x:
`hub fork ` will print 
> existing remote: pmjklemm

and a fork of brew-science is produced.

Here the full log:

> (base) [paulklemm@pk:~]$ cd $(brew --repo brewsci/science)
> (base) [paulklemm@pk:homebrew-science]$ git checkout master
> Switched to branch 'master'
> Your branch is up to date with 'origin/master'.
> (base) [paulklemm@pk:homebrew-science]$ git branch -D migrate-proteinortho
> Deleted branch migrate-proteinortho (was 27b86ed).
> (base) [paulklemm@pk:homebrew-science]$ git fetch origin
> (base) [paulklemm@pk:homebrew-science]$ git reset --hard origin/master
> HEAD is now at dcf7758 wiggletools: migrate to brewsci/bio (#241)
> (base) [paulklemm@pk:homebrew-science]$ hub fork
> existing remote: pmjklemm
> (base) [paulklemm@pk:homebrew-science]$ git fetch pmjklemm
> Warning: Permanently added the RSA host key for IP address '140.82.118.4' to the list of known 
> hosts.
> git@github.com: Permission denied (publickey).
> fatal: Could not read from remote repository.
> 
> Please make sure you have the correct access rights
> and the repository exists.
> Please make sure you have the correct access rights
> and the repository exists.
> (base) [paulklemm@pk:homebrew-science]$ brew migrate-formula --tap=homebrew/core brewsci/science/proteinortho
> ==> Migrating brewsci/science/proteinortho to homebrew/core
> Warning: Skipping new formula PR because formula already exists: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/proteinortho.rb
> Already on 'master'
> Your branch is up to date with 'origin/master'.
> Already up to date.
> Switched to a new branch 'migrate-proteinortho'
> rm 'Formula/proteinortho.rb'
> [migrate-proteinortho 9b97dbc] proteinortho: migrate to homebrew/core
>  2 files changed, 1 insertion(+), 35 deletions(-)
>  delete mode 100644 Formula/proteinortho.rb
> fatal: 'paulklemm' does not appear to be a git repository
> fatal: Could not read from remote repository.
> 
> Please make sure you have the correct access rights
> and the repository exists.
> Error: Failure while executing; `git push paulklemm migrate-proteinortho` exited with 128.
> 

So for `git fetch pmjklemm` it fails with some permission denies?
Not sure why there is a paulklemm ? my username for github should be pmjklemm.
